wandb/sdk/launch/_project_spec.py

```diff
@@ -139,8 +141,10 @@ def init_source(self) -> None:
         elif self.job is not None:
             self.source = LaunchSource.JOB
             self.project_dir = tempfile.mkdtemp()
-        else:
-            raise LaunchError("Launch project must have a source.")
+        if self.uri and self.uri.startswith("placeholder"):
+            self.source = LaunchSource.SCHEDULER
+            self.project_dir = os.getcwd()
+            self._entry_point = self.override_entrypoint
```

Comment:
Why set the entrypoint to override entrypoint?

Reply:
I was trying to accommodate [this](https://github.com/wandb/wandb/blob/531d06c2db15b50220c36dc807cff9734e3d9a60/wandb/sdk/launch/_project_spec.py#L341)

We could instead modify that error case to permit `LaunchSource.Scheduler`